proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,340 @@
+date: 2021-08-03
+---
+
+# RFC: Reduce API fragmentation across platforms
+
+## Summary
+
+This is a proposal to incrementally reduce the API fragmentation faced by developers using React Native to target multiple platforms, and sharing code between native and web. Each proposed change is additive and can be tackled relatively independently. The aim is to avoid needing to migrate existing React Native code, and to make deprecations an optional and future part of this work.
+
+## Motivation
+
+React Native currently includes many APIs that are modelled on Web APIs but do not conform to the standards of those Web APIs. React Native also includes many APIs that achieve the same results on Android and iOS but are exposed as 2 different props. And React Native includes several APIs that have known performance (network and runtime) drawbacks on Web.
+
+This proposal aims to allow developers to target more platforms with cross-platform APIs, and deliver better performance when targeting browsers. Features for Android, iOS, and Web are unified by aligning with the Web standard to a) minimize the overhead when running in browsers, and b) reduce developer education required to learn these features.
+
+## Detailed design
+
+Many of these additions are already in progress and have been proven in a user-space shim at Meta. The features that cannot be shimmed are explicitly called out below. Those features require core changes to React Native. Other features can be shimmed in user-space JavaScript if desired (with any associated performance cost of doing so).
+
+### New common props
+
+Certain props common to all core components have direct equivalents in React DOM. We should support those props where possible. For example, many of the existing `accessibility*` props in React Native are currently mapped to `aria-*` equivalents by React Native for Web. Supporting the ARIA interface would allow more ARIA-targeting React DOM libraries to work on React Native.
+
+* ARIA props
+  * Support `aria-*` aliases for `accessibility*` props.
+  * `aria-busy` => `accessibilityState.busy`
+  * `aria-checked` => `accessibilityState.checked`
+  * `aria-disabled` => `accessibilityState.disabled`
+  * `aria-expanded` => `accessibilityState.expanded`
+  * `aria-hidden` => `accessibilityElementsHidden` (iOS) + `importantforAccessibility` (Android)
+  * `aria-label` => `accessibilityLabel`
+  * `aria-live` => `accessibilityLiveRegion` (and add `'off'` value => `'none'`)
+  * `aria-modal` => `accessibilityViewIsModal`
+  * `aria-selected` => `accessibilityState.selected`
+  * `aria-valuemax` => `accessibilityValue.max`
+  * `aria-valuemin` => `accessibilityValue.min`
+  * `aria-valuenow` => `accessibilityValue.now`
+  * `aria-valuetext` => `accessibilityValue.text`
+  * `role` => `accessibilityRole`
+  * See prior work: [RFC Accessibility APIs](https://github.com/react-native-community/discussions-and-proposals/pull/410/files).
+* `id`
+  * Add `id` prop as synonym for `nativeID`.
```

Comment:
is the use of the word 'synonym' instead of 'alias' here (and in several other places in this rfc) intentional? if so, can you clarify the difference in meaning?